realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -508,6 +548,22 @@ private boolean categorizeField(Element element) {
         return true;
     }
 
+    private String getInternalFieldName(VariableElement field, NameConverter defaultConverter) {
+        RealmField nameAnnotation = field.getAnnotation(RealmField.class);
+        if (nameAnnotation != null) {
+            String declaredName = nameAnnotation.name();
+            if (!declaredName.equals("")) {
```

Comment:
We actually moved all of that logic to OS instead of having it in the annotation processor, but having the check will not only improve the error message, but also throw on compile time. I'll add it.

Reply:
Actually, in this place. It is just field names. Field names do not have any restrictions